packages/manager/cypress/e2e/core/account/change-username.spec.ts

```diff
@@ -96,4 +110,57 @@ describe('username', () => {
       cy.findByText('Username updated successfully.').should('be.visible');
     });
   });
+
+  it('disables username/email fields for proxy user', () => {
+    const mockRestrictedProxyUser = accountUserFactory.build({
+      restricted: true,
+      user_type: 'proxy',
+      username: 'restricted-proxy-user',
+    });
+
+    const mockRestrictedProxyProfile = profileFactory.build({
+      username: 'restricted-proxy-user',
+      user_type: 'proxy',
+    });
+
+    const mockUserGrants = grantsFactory.build({
+      global: { account_access: 'read_write' },
+    });
```

Comment:
```suggestion
    const mockRestrictedProxyProfile = profileFactory.build({
      username: 'restricted-proxy-user',
      user_type: 'proxy',
      restricted: true
    });
```

I don't think we need to mock the account users endpoint here at all, nor grants. We [disable the buttons and display the tooltips](https://github.com/linode/manager/blob/develop/packages/manager/src/features/Profile/DisplaySettings/DisplaySettings.tsx#L107-L114) based on whether the user is restricted or a proxy user. We can use `restricted` and `user_type` from `*/profile` for that.